staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -242,9 +241,27 @@ func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision
 	if len(history) <= 0 {
 		return "No rollout history found.", nil
 	}
-	revisions := make([]int64, 0, len(history))
-	for _, revision := range history {
-		revisions = append(revisions, revision.Revision)
+
+	historyInfo := make(map[int64][]byte)
+	for _, h := range history {
+		historyInfo[h.Revision] = h.Data.Raw
+	}
+	if revision != 0 {
+		value, ok := historyInfo[revision]
+		if !ok {
+			return "", fmt.Errorf("unable to find the specified revision")
+		}
+		sts := appsv1.StatefulSet{}
+		err := json.Unmarshal(value, &sts)
+		if err != nil {
+			return "", fmt.Errorf("error unmarshelling sts: %w", err)
+		}
+		return printTemplate(&sts.Spec.Template)
```

Comment:
Are we only interested in printing template? Are we intentionally ignoring `volumeClaimTemplates` and `StatefulSet`'s other fields?

Reply:
Hi @m1kola Thanks for the review. In sts, we *cannot update* fields other than 'replicas', 'template', and 'updateStrategy'. Editing `replicas` and `updateStrategy` will not create new revisions. So I guess showing only the pod template during rollout history is okay. What do you think? Please provide your input

Reply:
Thanks for the clarification. I'll mark the PR as ok to test, but I'm not sure about this bit. Would be great to have someone else's opinion on this.

Reply:
Note that changing `volumeClaimTemplates` is not supported, but we may support volume expansion https://github.com/kubernetes/enhancements/issues/661